src/core/app-template.cc

```diff
@@ -126,6 +126,16 @@ app_template::configuration_reader app_template::get_default_configuration_reade
     };
 }
 
+void app_template::reread_config() {
+    bpo::variables_map configuration;
+    if (_configuration->contains("config-file")) {
+        configuration.emplace("config-file", (*_configuration)["config-file"]);
+    }
+    _conf_reader(configuration);
+    program_options::variables_map_extracting_visitor visitor(configuration);
+    _opts.mutate_live(visitor);
```

Comment:
i am wondering if we can just skip the reread if `!_configuration->contains("config-file")`? as there is no point of reading a conffile unless the file is specified in the command line or in the previous config file. so might want to put something like:
```c++
auto config_file = _configuration->find("config-file");
if (config_file == _configuration->end()) {
    return;
}
// reread on demand
```

please note, in order to avoid double-lookup, i am using `find()` here. not a big deal, but it also helps remove the one more repeating of "`config-file`".

Reply:
> i am wondering if we can just skip the reread if !_configuration->contains("config-file")? as there is no point of reading a conffile unless the file is specified in the command line or in the previous config file

HUP may still mean that the default $HOME/.config/seastar/seastar.conf was updated and needs to be re-read.

Reply:
ahh, right!